automate/automate-templates/repository.txt

```diff
@@ -22,8 +22,18 @@ func New{{ucresource}}Repository(db infrastructure.Database, logger infrastructu
 	}
 }
 
-// Save {{ucresource}}
-func (c {{ucresource}}Repository) Save({{ucresource}} models.{{ucresource}}) (models.{{ucresource}}, error) {
+// WithTrx enables repository with transaction
+func (tx {{ucresource}}Repository) WithTrx(trxHandle *gorm.DB) {{ucresource}}Repository {
+	if trxHandle == nil {
+		tx.logger.Zap.Error("Transaction Database not found in gin context. ")
+		return tx
+	}
+	tx.db.DB = trxHandle
+	return tx
+}
+
```

Comment:
Please add transaction in service template also otherwise we cant pass db instance from 
controller to repo. 

I would rather suggest not to use transaction function in templates.
We have given example in user how to use transaction and  transation wont be needed in all the apis.

Reply:
> Please add transaction in service template also otherwise we cant pass db instance from controller to repo.
> 
> I would rather suggest not to use transaction function in templates. We have given example in user how to use transaction and transaction wont be needed in all the apis.

@viveksharmapoudel 
Fixed 🙇